From 742aec35da9a7f1c2fa98423b4b3f572baffc2f8 Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:22:02 -0400
Subject: [infiniband] ocrdma: Remove hardcoding of the max DPP QPs supported

Message-id: <41f0d3e7f6a825e5e1140994ce41832ddaf86ac7.1411571481.git.dledford@redhat.com>
Patchwork-id: 96626
O-Subject: [Patch RHEL7 27/48] RDMA/ocrdma: Remove hardcoding of the max DPP QPs supported
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

Removing hardcoded value of max dpp qps and calculate the same from
doorbell page size and WQE size.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Selvin Xavier <selvin.xavier@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit a53d77a33494a3d1d5c797df311c975a05d69ef3)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_sli.h b/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
index 4defae8..14a84b2 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_sli.h
@@ -1236,7 +1236,6 @@ struct ocrdma_destroy_srq {
 
 enum {
  OCRDMA_ALLOC_PD_ENABLE_DPP = BIT(16),
- OCRDMA_PD_MAX_DPP_ENABLED_QP = 8,
  OCRDMA_DPP_PAGE_SIZE  = 4096
 };
 
diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
index edf6211..0d7d808 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_verbs.c
@@ -268,7 +268,8 @@ static struct ocrdma_pd *_ocrdma_alloc_pd(struct ocrdma_dev *dev,
   pd->dpp_enabled =
    ocrdma_get_asic_type(dev) == OCRDMA_ASIC_GEN_SKH_R;
   pd->num_dpp_qp =
-   pd->dpp_enabled ? OCRDMA_PD_MAX_DPP_ENABLED_QP : 0;
+   pd->dpp_enabled ? (dev->nic_info.db_page_size /
+        dev->attr.wqe_size) : 0;
  }
 
 retry:
-- 
1.7.1